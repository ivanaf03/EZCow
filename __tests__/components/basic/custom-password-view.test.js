import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import CustomPasswordView from "../../../components/basic/custom-password-view";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
    FontAwesomeIcon: () => null
}));

describe("CustomPasswordView", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(
            <CustomPasswordView
                text={'test'}
                value={'test'}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    it("should show password when clicking on eye icon", () => {
        const onChangeText = jest.fn();
        const tree = render(
            <CustomPasswordView
                text={'test'}
                value={'test'}
                onChangeText={onChangeText}
            />
        );
        const eyeIcon = tree.getByTestId("eye-icon");
        expect(tree.getByTestId("password-text").props.children).toBe("*".repeat(4));        
        fireEvent.press(eyeIcon);
        expect(tree.getByTestId("password-text").props.children).toBe("test");
    });

    it("should hide password when clicking on eye icon", () => {
        const onChangeText = jest.fn();
        const tree = render(
            <CustomPasswordView
                text={'test'}
                value={'test'}
                onChangeText={onChangeText}
            />
        );
        const eyeIcon = tree.getByTestId("eye-icon");
        expect(tree.getByTestId("password-text").props.children).toBe("*".repeat(4));
        fireEvent.press(eyeIcon);
        expect(tree.getByTestId("password-text").props.children).toBe("test");
        fireEvent.press(eyeIcon);
        expect(tree.getByTestId("password-text").props.children).toBe("*".repeat(4));
    });

});