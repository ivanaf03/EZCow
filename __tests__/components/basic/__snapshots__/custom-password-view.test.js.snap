// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPasswordView should render correctly 1`] = `
<View
  style={
    [
      {
        "marginTop": -4,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "fontFamily": "Nunito-Light",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    test
    :
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginRight": 8,
          },
          {
            "backgroundColor": "#272727",
          },
          {
            "color": "#ffffff",
          },
          {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
      testID="password-text"
    >
      ****
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="eye-icon"
    />
  </View>
</View>
`;
